### PR TITLE
Create simple startup script for admin development on linc-brain

### PR DESCRIPTION
**Pull request by aaronkanzer**
This PR creates `admin_dev_startup.sh` script to facilitate easy development for `linc-archive` locally via docker/docker-compose.

@kabilar -- see below

Notes:

1. The developer will be prompted in the middle of the script to enter in a username and password -- be sure to note this (see the step below for further context 😄 )

2. After running the script, the developer will need to go to `localhost:8000/admin` and authenticate themselves to use the local dev correctly. Otherwise, authentication will fail. The username/password should be the pair that the developer entered when running the startup script.

Furthermore, ensure that the user is `APPROVED` via the `Users` tab in the Django Admin console.

3. There is 'hot-reloading' for the UI here -- so as you make changes in the codebase for the frontend, you should see your changes automagically applied locally.

To try it out locally before we merge:

- Go to `linc-archive` locally

- Run `git status` to see if you have any changes to commit before switching branches (if so, don't go to the next steps until you are OK with the state of your repo

- Run `git fetch -a` to get all the remote branches

- Run `git checkout ak-ui-work` to switch to our branch

- Run `./admin_dev_startup.sh <fun-image-name-you-can-pick> <an-email-such-as-your-mit-email>`

- Stay tuned in the script and hope it works!

### CODE REVIEW

**Comment by kabilar** [comment]
Thanks @aaronkanzer.  This is great work.  I tested by making changes to the front-end text and the web app hot reloads correctly.

One minor suggestion would be to add a section to the readme with your instructions that I have summarized below.  Once this feature is integrated within the DANDI repo, we can move these instructions to the DANDI handbook.

```
## Instructions for local development with hot reloading
1. Install Docker
1. Run `./admin_dev_startup.sh <fun-image-name-you-can-pick> <your-email>`.  When prompted, enter an username and password in the command prompt.
1. Navigate to `localhost:8000/admin`, and log in with the username and password.  Under the `User` section, select the username and change the `Status` from `Incomplete` to `Approved`.
1. Navigate to `localhost:8085` and select `LOG IN WITH GITHUB`.
```

File: dandiapi/api/management/commands/create_dev_dandiset.py [hunk 1]
```diff
@@ -26,16 +32,19 @@ def create_dev_dandiset(*, name: str, email: str):
...
+    file_content = b'A' * file_size
+    uploaded_file = SimpleUploadedFile(name='foo/bar.txt', content=file_content)
     etag = '76d36e98f312e98ff908c8c82c8dd623-0'
+
     try:
         asset_blob = AssetBlob.objects.get(etag=etag)
     except AssetBlob.DoesNotExist:
+        # Since the SimpleUploadedFile is non-zarr asset, validation fails
+        # without a sha2_256 initially provided.
+        sha256_hash = hashlib.sha256(file_content).hexdigest()
```
Comment (kabilar) [inline dandiapi/api/management/commands/create_dev_dandiset.py:45]: I am not quite familiar with this piece here, but we can proceed with merging and revisit once we get feedback from other members of the DANDI team in https://github.com/dandi/dandi-archive/pull/1809.

File: admin_dev_startup.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Run the Docker container for frontend in background
+docker run -d -v "$(pwd):/usr/src/app" -v /usr/src/app/node_modules -p 8085:8085 -e CHOKIDAR_USEPOLLING=true $image_name
+
+cd ..
+
+# Run Docker Compose commands for backend
+docker-compose run --rm django ./manage.py migrate
+docker-compose run --rm django ./manage.py createcachetable
+docker-compose run --rm django ./manage.py createsuperuser
+docker-compose run --rm django ./manage.py create_dev_dandiset --owner $email
```
Comment (kabilar) [inline admin_dev_startup.sh:28]: Would there be any issues with explicitly switching to Compose V2?

```suggestion
docker compose run --rm django ./manage.py migrate
docker compose run --rm django ./manage.py createcachetable
docker compose run --rm django ./manage.py createsuperuser
docker compose run --rm django ./manage.py create_dev_dandiset --owner $email
```

File: admin_dev_startup.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# Bring backend application to life!
+docker-compose up
```
Comment (kabilar) [inline admin_dev_startup.sh:31]:
```suggestion
docker compose up
```

File: admin_dev_startup.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+# This script can be used to launch the UI with ease locally
```
Comment (kabilar) [inline admin_dev_startup.sh:3]:
```suggestion
# This script can be used to launch the front-end and back-end applications locally.
```

**Review by kabilar — CHANGES_REQUESTED** [review]
Thank you @aaronkanzer.  A few minor comments above.

File: admin_dev_startup.sh (hunk 4 above)
Comment (aaronkanzer) [reply admin_dev_startup.sh]: Updated!

File: admin_dev_startup.sh (hunk 3 above)
Comment (aaronkanzer) [reply admin_dev_startup.sh]: Updated!

File: admin_dev_startup.sh (hunk 2 above)
Comment (aaronkanzer) [reply admin_dev_startup.sh]: Updated!

**Review by kabilar — APPROVED** [review]
Thank you @aaronkanzer.  Great work.

**Pull request merged**